File: services/auth-service/src/main/java/com/makestar/authservice/service/AuthService.java
package com.makestar.authservice.service;

import com.makestar.dto.auth.LoginRequestDto;
import com.makestar.dto.auth.LoginResponseDto;
import com.makestar.dto.user.UserDto;

public interface AuthService {
    
    /**
     * 사용자 로그인 처리
     * @param loginRequest 로그인 요청 정보
     * @return 로그인 응답 (토큰, 사용자 정보 등)
     */
    LoginResponseDto login(LoginRequestDto loginRequest);
    
    /**
     * 사용자 회원가입 처리
     * @param userDto 회원가입 정보
     * @return 생성된 사용자 정보
     */
    UserDto register(UserDto userDto);
    
    /**
     * 리프레시 토큰을 사용하여 액세스 토큰 갱신
     * @param refreshToken 리프레시 토큰
     * @return 로그인 응답 (새 토큰, 사용자 정보 등)
     */
    LoginResponseDto refreshToken(String refreshToken);
    
    /**
     * 토큰 유효성 검증
     * @param token 검증할 토큰
     * @return 유효 여부
     */
    boolean validateToken(String token);
} 

File: frontend/web-client/src/services/authService.ts
import api from './api';

// 타입 정의
export interface LoginRequest {
  username: string;
  password: string;
}

export interface RegisterRequest {
  username: string;
  email: string;
  password: string;
}

export interface AuthResponse {
  token: string;
  user: {
    id: number;
    username: string;
    email: string;
  };
}

// 인증 서비스
const authService = {
  // 로그인
  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/login', data);
    return response.data;
  },
  
  // 회원가입
  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/register', data);
    return response.data;
  },
  
  // 사용자 정보 가져오기
  async getUserInfo(): Promise<any> {
    const response = await api.get('/auth/me');
    return response.data;
  },
  
  // 로그아웃
  async logout(): Promise<void> {
    await api.post('/auth/logout');
    localStorage.removeItem('token');
  }
};

export default authService; 